src/openmrs-backend-dependencies.ts

```diff
@@ -0,0 +1,92 @@
+import { openmrsFetch } from "@openmrs/esm-api";
+import * as semver from "semver";
+import { difference, isEmpty } from "lodash-es";
+
+var installedBackendModules: any[] = [];
+
+const originalOnload = System.constructor.prototype.onload;
+
+System.constructor.prototype.onload = function(err, id, deps) {
+  if (!err) {
+    System.import(id).then(module => {
+      console.log(module);
+      if (module.backendDependencies) {
+        checkBackendDeps(module.backendDependencies);
+      }
+    });
+  }
+  return originalOnload.apply(this, arguments);
+};
+
+function checkBackendDeps(backendModules: any) {
+  if (!isEmpty(installedBackendModules)) {
+    fetchInstalledBackendModules().then(({ data }) => {
+      installedBackendModules = data.results;
+      checkIfModulesAreInstalled(backendModules);
+    });
+  } else {
+    checkIfModulesAreInstalled(backendModules);
+  }
+}
+
+function checkIfModulesAreInstalled(requiredBackendModules) {
+  const missingOpenmrsBackendModules = getMissingBackendModules(
+    requiredBackendModules
+  );
+  const modulesWithWrongVersionInstalled = getMismatchedVersions(
+    requiredBackendModules
+  );
+  console.error(
+    "The following backend modules are required!",
+    missingOpenmrsBackendModules
+  );
+  console.error(
```

Comment:
Should we only console.log if the `modulesWithWrongVersionInstalled` array has a length > 0? Same goes for the `missingOpenmrsBackendModules` console error